SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/config/SingularityS3UploaderConfiguration.java

```diff
@@ -42,15 +43,19 @@
   @Obfuscate
   private Optional<String> s3SecretKey = Optional.absent();
 
+  @Max(5368709120L)
   @JsonProperty
   private long maxSingleUploadSizeBytes = 5368709120L;
 
+  @Min(5242880L)
   @JsonProperty
   private long uploadPartSize = 20971520L;
 
+  @Min(500)
```

Comment:
how'd you decide on 500? probably fine, but maybe we should just make the minimum 0 so there's no arbitrary cutoff